### ISSUE TITLE
std::tuple_size<T> with emp::array

### ISSUE THREAD

**Issue by mmore500** [devosoft/Empirical#247]
The standard library template [`std::tuple_size<class T>`](https://en.cppreference.com/w/cpp/utility/tuple/tuple_size) likes `std::array` but doesn't play nice with `emp::array`, yielding a compile-time error.

**Comment by mmore500** [comment]
Minimal failing example
```
#include <tuple>
#include <iostream>

#include "base/array.h"

int main() {
  using arr_t = emp::array<int, 3>;
  std::cout << std::tuple_size<arr_t>::value << std::endl;
}
```

**Comment by mmore500** [comment]
compiler error message
```
test_bs.cc: In function ‘int main()’:
test_bs.cc:8:40: error: incomplete type ‘std::tuple_size<emp::array<int, 3> >’ used in nested name specifier
   std::cout << std::tuple_size<arr_t>::value << std::endl;
                                        ^~~~~
```

**Comment by mmore500** [comment]
Closed by #274
